operations/api_operation_test.go

```diff
@@ -0,0 +1,66 @@
+package operations
+
+import (
+	"bytes"
+	"io"
+	"testing"
+)
+
+type MockNormalApiOperationApiClient struct{}
+
+func (client MockNormalApiOperationApiClient) Request(method string, path string, body io.Reader) ([]byte, error) {
+	return bytes.NewBufferString("{\"say\":\"hello\"}").Bytes(), nil
+}
+
+func TestExecutingApiOperation(t *testing.T) {
+	client := &MockNormalApiOperationApiClient{}
+	oper := NewApiOperation("GET", "https://somewhere", bytes.NewBufferString(""), client)
+	result := oper.run()
+
+	if result.is_error != false {
+		t.Errorf("Expected no error to be returned.")
+	}
+}
+
+func ExampleApiOperation_run_output() {
+	client := &MockNormalApiOperationApiClient{}
+	oper := NewApiOperation("GET", "https://somewhere", bytes.NewBufferString(""), client)
+
+	oper.run()
+	// Output:
+	// {
+	//   "say": "hello"
+	// }
```

Comment:
COMMENT: It's interesting for me 😄 (I thought this is just comment once, but I realized it's a matcher by googling...)